### PR TITLE
Full es-module support

### PR DESCRIPTION
**Pull request by kentandersen**
TLDR; CommonJS output is unchanged. Esm output has proper `.js` also for type definition files, Jest has been replaced with mocha

There is an issue when using the es-module version. The generated types cannot be found. The culprit is missing .js extension in import in `dist/esm/types/index.d.ts`. I have added `.js` to all source files to properly support es-modules. 

This fix breaks a workaround [we did earlier](https://github.com/entur/reduce-customers-for-offer-configuration/pull/5) which adds `.js` to the esm output after transportation. The reason for this workaround was that Jest would not play nice with es-modules. Jest have only [experimental (not working) support for es-modules](https://jestjs.io/docs/ecmascript-modules).

Mocha is a leaner spec-runner, which can be configured to support es-modules. I have replaced jest with mocha. Only real difference is that `expect` has to be imported, and `test()` is replaced with `it()`.
Esm does not support importing json data, so I've added a `readJSON()` helper.

### CODE REVIEW

File: scripts/build-commonjs.sh [hunk 1]
```diff
@@ -2,14 +2,7 @@
 
 npx tsc --module commonjs --target es5 --outDir dist/commonjs
 
-SED_OPTIONS=
```
Comment (kentandersen) [inline scripts/build-commonjs.sh:5]: The replace inline `-i ''` sed argument for macOS added '' to the files, instead of leaving the file names untouched. New approach is os independent, rendering the `$SED_OPTIONS` redundant.

**Comment by frankwaldal** [comment]
Ran `build` and `test` locally on Linux, no issues. Also tested the package locally in Sørvis after new build. All good! :+1:

**Review by joelchelliah — APPROVED** [review]

File: test/helper.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import {readFile} from 'node:fs/promises';
```
Comment (Vages) [inline test/helper.ts:1]: Supert med bruk av node-protokollen 😁
